monix-execution/shared/src/main/scala/monix/execution/FutureUtils.scala

```diff
@@ -115,18 +116,28 @@ object FutureUtils {
     *
     * Similar to `Future.transformWith` from Scala 2.12.
     */
-  def transformWith[A,B](source: Future[A], f: Try[A] => Future[B])(implicit ec: ExecutionContext): Future[B] = {
+  def transformWith[A,B](source: Future[A], f: Try[A] => Future[B])(implicit ec: ExecutionContext): CancelableFuture[B] = {
```

Comment:
Yeah, it helps. I'll have to rework this change to make it binary compatible.

@alexandru WDYT?